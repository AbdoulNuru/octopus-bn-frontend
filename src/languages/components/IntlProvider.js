import React, { Component } from 'react';
import { IntlProvider } from 'react-intl';
import { connect } from 'react-redux';

class Provider extends Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<IntlProvider locale={this.props.languageState.locale} messages={this.props.languageState.messages}>
				{this.props.children}
			</IntlProvider>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		languageState: state.language
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		setLanguage: (language) => dispatch(loginAction(language))
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(Provider);
