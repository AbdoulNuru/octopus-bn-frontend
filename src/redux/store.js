import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import rootReducer from './reducers';
import logger from 'redux-logger';
const middlwares = applyMiddleware(thunk, promise, logger);
const store = createStore(rootReducer, {}, composeWithDevTools(middlwares));

export default store;
