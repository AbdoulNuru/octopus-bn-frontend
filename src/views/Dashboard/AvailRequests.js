import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import FlightTakeoffIcon from '@material-ui/icons/FlightTakeoff';
import styles from '../../styles/dashboard.requests.module.css';
import AvailRequestsTable from '../../components/AvailRequestsTable';
import { connect } from 'react-redux';
import { getRequestsAction } from '../../redux/actions/AvailRequestsActions';
import { withRouter } from 'react-router-dom';

class AvailRequests extends Component {
	componentDidMount() {
		this.props.getRequestsAction();
	}
	render() {
		return (
			<div>
				<div className={styles.topIntro}>
					<div className={styles.left}>
						<p className={styles.title}>Dashboard - AVAIL REQUESTS</p>
						<p className={styles.sub_title}>Travel made easy - Barefoot Nomad</p>
					</div>
					<div className={styles.right}>
						<Button
							variant="contained"
							className={styles.button}
							color="primary"
							startIcon={<FlightTakeoffIcon />}
						>
							Request Travel
						</Button>
					</div>
				</div>
				<div className={styles.table}>
					<AvailRequestsTable />
				</div>
			</div>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		loginSate: state.login
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		getRequestsAction: () => dispatch(getRequestsAction())
	};
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(AvailRequests));
