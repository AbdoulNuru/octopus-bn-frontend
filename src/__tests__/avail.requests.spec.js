import React from 'react';
import { render, fireEvent, cleanup, waitFor, waitForElement } from '@testing-library/react';
import AvailRequests from '../views/Dashboard/AvailRequests';
import Svg from '../components/SvgMap';
import { Provider } from 'react-redux';
import store from '../redux/store';
import userEvent from '@testing-library/user-event';
import IntlProvider from '../languages/components/IntlProvider';
import { createMemoryHistory } from 'history';
import { BrowserRouter as Router, Link } from 'react-router-dom';

const history = createMemoryHistory();
const AvailRequestsComponent = () => {
	return render(
		<Provider store={store}>
			<IntlProvider>
				<Router>
					<AvailRequests />
				</Router>
			</IntlProvider>
		</Provider>
	);
};
describe('AVAIL REQUEST', () => {
	afterEach(cleanup);
	it('should render login component', () => {
		const { asFragment } = AvailRequestsComponent();
		expect(asFragment(<AvailRequests />)).toMatchSnapshot();
	});
	it('should render login component', () => {
		const {
			getByLabelText,
			getByDisplayValue,
			getByText,
			getByTestId,
			container,
			debug
		} = AvailRequestsComponent();
		const next = getByLabelText('Next page');
		const prev = getByLabelText('Previous page');
		const row = getByLabelText('Rows per page:');
		fireEvent.click(next);
		fireEvent.click(prev);

		fireEvent.change(getByDisplayValue(/5/i), {
			target: { value: '10' }
		});
		expect(getByDisplayValue(/10/i)).toBeTruthy();
	});
});
