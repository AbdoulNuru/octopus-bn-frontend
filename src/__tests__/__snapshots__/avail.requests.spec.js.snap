// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AVAIL REQUEST should render login component 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <p>
          Dashboard - AVAIL REQUESTS
        </p>
        <p>
          Travel made easy - Barefoot Nomad
        </p>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2.5 19h19v2h-19v-2zm19.57-9.36c-.21-.8-1.04-1.28-1.84-1.06L14.92 10l-6.9-6.43-1.93.51 4.14 7.17-4.97 1.33-1.97-1.54-1.45.39 2.59 4.49s7.12-1.9 16.57-4.43c.81-.23 1.28-1.05 1.07-1.85z"
                />
              </svg>
            </span>
            Request Travel
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div>
      <div
        class="MuiPaper-root makeStyles-root-42 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <div
            class="MuiTypography-root MuiTypography-h6"
            id="tableTitle"
          >
            Avail requests
          </div>
        </div>
        <table
          aria-label="caption table"
          class="MuiTable-root makeStyles-table-43"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Type
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                By
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                From
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                To
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                Departure Date
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                Return Date
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                colspan="6"
              >
                <div
                  class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                  role="progressbar"
                  style="width: 24px; height: 24px;"
                >
                  <svg
                    class="MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleDisableShrink MuiCircularProgress-circleIndeterminate"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20"
                      stroke-width="4"
                    />
                  </svg>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
          >
            <div
              class="MuiTablePagination-spacer"
            />
            <p
              class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
            >
              <div
                aria-haspopup="listbox"
                aria-label="Rows per page:"
                class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                role="button"
                tabindex="0"
              >
                5
              </div>
              <input
                type="hidden"
                value="5"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            >
              0-0 of 0
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Previous page"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                title="Previous page"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Next page"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                title="Next page"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
